pkg/loki/query.go

```diff
@@ -143,7 +147,7 @@ func (q *Query) AddParam(key, value string) error {
 	case matchParam, flowDirParam:
 		q.specialAttrs[key] = value
 	// Stream selector labels
-	case "SrcNamespace", "SrcWorkload", "DstNamespace", "DstWorkload":
+	case utils.GetMatchingString(q.labels, key):
```

Comment:
I'm afraid that doing this could modify the underlying implementation of the `switch` from a map-based lookup to a chain of if-elses. Maybe we could try benchmarking this option vs an option where this case is moved to the `default`, also changing the array loop by a map lookup:

```go
default:
  if _, ok := q.labelMap[key]; ok {
    q.processStreamSelector(key, strings.Split(value, ","))
  } else {
    return q.processLineFilters(key, strings.Split(value, ","))
  }

Reply:
Sure, I was a bit disappointed we can write multiple strings in the same case but can't provide an array